### PR TITLE
Added support for new types

### PR DESCRIPTION
**Pull request by YPermitin**
- #32 Added support for Uri type
- #39 Added test for ReadOnlyCollection<> types.
- #27 Added support for Enum type
- Other small fixes.

### CODE REVIEW

File: src/BinaryFormatter/TypeConverter/GuidConverter.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ protected override Guid ProcessDeserialize(byte[] stream, Type sourceType, ref i
 
         protected override int GetTypeSize()
         {
-            return sizeof (byte);
+            return 16;
```
Comment (lukasz-pyrzyk) [inline src/BinaryFormatter/TypeConverter/GuidConverter.cs]: yup! just saw that

**Comment by lukasz-pyrzyk** [comment]
Does EnumConverter supports enums with different sizes? Like enum : short or byte? What about flags?

File: src/BinaryFormatter/TypeConverter/GuidConverter.cs (hunk 1 above)
Comment (YPermitin) [inline src/BinaryFormatter/TypeConverter/GuidConverter.cs]: @lukasz-pyrzyk Oooops...
I will fix it!

**Comment by YPermitin** [comment]
@lukasz-pyrzyk Absolutly. But I will add tests for all sizes of enum.
I think that will do it today.

**Comment by YPermitin** [comment]
@lukasz-pyrzyk Also I will update the readme

**Comment by lukasz-pyrzyk** [comment]
Great :)

**Comment by YPermitin** [comment]
@lukasz-pyrzyk I did it. Please check fixes in GuidConverter and EnumConverterTests.

**Comment by lukasz-pyrzyk** [comment]
Nice! Please update readme and i am happy to merge

**Comment by YPermitin** [comment]
Ready!

**Pull request merged**